tuner/tuner/libtuner.py

```diff
@@ -1045,23 +1054,21 @@ def benchmark(
         logging.warning("Baseline run failed.")
 
     speedup_result = baseline_handler.calculate_speedup(candidate_results)
-    # If the baseline is valid (`baseline_handler.is_valid()`), `speedup_result` represents the speedup values.
-    # Otherwise, `speedup_result` contains the raw time values.
-    top_candidates = baseline_handler.get_top_candidates(speedup_result, num_candidates)
+    all_candidates_with_speedup = baseline_handler.get_top_candidates(speedup_result)
```

Comment:
Speedup result should be calculated inside get_top_candidates. We should also rename this to something else now that it doesn't produce top N values, and instead, returns the full list. See the full suggestion here: https://github.com/nod-ai/shark-ai/pull/789#discussion_r1920717256